### ISSUE TITLE
about adding BatchNorm

### ISSUE THREAD

**Issue by nnop** [nnop/notes#102]
- https://github.com/revilokeb/vgg16_batchnorm
- https://github.com/HolmesShuan/AlexNet-BN-Caffemodel-on-ImageNet

**Comment by nnop** [comment]
- [Understanding the backward pass through Batch Normalization Layer](https://kratzert.github.io/2016/02/12/understanding-the-gradient-flow-through-the-batch-normalization-layer.html)
- cs231n assignment 2

**Comment by nnop** [comment]
fine-tune BatchNorm

- vlfeat/matconvnet issue: 533

**Comment by nnop** [comment]
- [On The Perils of Batch Norm](http://www.alexirpan.com/2017/04/26/perils-batch-norm.html)
- [Batch normalization - 95.9%](http://rizar.github.io/ift6266h16/jekyll/update/2016/04/18/batch-normalization.html)

**Comment by nnop** [comment]
Caffe BatchNorm:

- https://github.com/BVLC/caffe/issues/3347#issuecomment-191068006
